corporate/lib/stripe_event_handler.py

```diff
@@ -150,3 +152,34 @@ def handle_payment_intent_succeeded_event(
         False,
         billing_session.get_remote_server_legacy_plan(payment_intent.customer),
     )
+
+
+@error_handler
+def handle_invoice_paid_event(stripe_invoice: stripe.Invoice, invoice: Invoice) -> None:
+    invoice.status = Invoice.PAID
+    invoice.save(update_fields=["status"])
+
+    customer = invoice.customer
+
+    configured_fixed_price_plan = None
+    if customer.required_plan_tier:
+        configured_fixed_price_plan = get_configured_fixed_price_plan_offer(
+            customer, customer.required_plan_tier
+        )
```

Comment:
This logic is fine for now, but I think we should still be planning to move `required_plan_tier` to be a `tier` field of `CustomerPlanOffer`.

Reply:
Yeah, planning to do this next to prevent more such additions.